students/johnpharmd/lesson03/seqslicing.py

```diff
@@ -0,0 +1,104 @@
+def seqxchg(seq):
+    """
+    This function returns a copy of the sequence passed but with
+    the first and last items exchanged.
+    """
+    # new_seq = [seq[-1], seq[1:-1], seq[0]] -- this failed
```

Comment:
I'm glad that you left this comment, because you were close..

let's break it down:
let's say seq is `[1, 2, 3, 4]`
[seq[-1], seq[1:-1], seq[0]] would return: `[4, [2, 3], 1]` not really what we want right? as you notice accessing list by an index returns a single item, but slicing returns another list. So let's stick to slicing only and use `+` to add the list parts together

so we get:
`seq[-1:] + seq[1:-1] + seq[:1]`

Reply:
also for this exercise, there should be no difference in slicing logic for list, string or a tuple so type checking is not necessary.

a side note, it is preferred to use `isinstance` method instead of `type` to check the type when you _do_ need to.